### PR TITLE
Minor refactor of array_position

### PR DESCRIPTION
**Pull request by rongrong**
* Remove redundant logic to check null on equal method invocation
* Clarify error message
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java [hunk 1]
```diff
@@ -158,7 +159,9 @@ public static long arrayPosition(
                 Object arrayValue = type.getObject(array, i);
                 try {
                     Boolean result = (Boolean) equalMethodHandle.invoke(arrayValue, element);
-                    checkNotIndeterminate(result);
+                    if (result == null) {
+                        throw new PrestoException(NOT_SUPPORTED, "array_position does not support element of array type that contains null");
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java]: Maybe something like `array_position does not support elements of complex types that contain null`? In theory `T` can be a map or a row.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java [hunk 2]
```diff
@@ -131,7 +132,7 @@ public static long arrayPosition(
                 Slice arrayValue = type.getSlice(array, i);
                 try {
                     Boolean result = (Boolean) equalMethodHandle.invokeExact(arrayValue, element);
-                    checkNotIndeterminate(result);
+                    checkArgument(result != null, "Array element should not be nulll");
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java]: nit: This check rather looks like an assertion. No matter what arguments are passed to this function the equals function should never return `null`, as the `null` case is explicitly checked before. For such assertions usually `verify` is preferred. (same for others).

Comment (rongrong) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java]: InteliJ would give warning on `if (result)` about null pointer exception when using `verify`. It's slightly annoying to look at but that's fine I suppose.

Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayPositionFunction.java]: It is interesting why Intellij understands `checkArgument` but doesn't undestand `verify` =\

**Pull request merged**